lib/fizzy/instantiate.hpp

```diff
@@ -21,8 +21,31 @@ namespace fizzy
 struct ExecutionResult;
 struct Instance;
 
+class ThreadContext
+{
+public:
+    int depth = 0;
```

Comment:
So there's no way to set a different starting depth?

Reply:
You can `ctx.depth = 1`, as in tests.

Reply:
Should we expose it? In any case if we it should be just a separate PR for the C API (and then Rust). It is fine to hide it in this PR.

Reply:
> In C the ThreadContext is opaque what is good for API, but it is impossible to dump the depth inside a C host function.

Reply:
Well we would need C++ helpers for it first, right? :)